### PR TITLE
Adds the "Unlock Antag HUD" verb to Ghosts that also blocks them from coming back into the round

### PR DESCRIPTION
**Pull request by Wayland-Smithy**
## About The Pull Request

Before you go to the comments I ask you to consider that for the situation of a "metacomms observer" this PR would change little to nothing other than prevent that person from coming into the round themselves which is a net positive in my book.

Someone joining or staying after death for the sole purpose of cheating for their metafriends may be able to acquire their info a little faster sure, but between the observe verb, ability to look in containers, and dchat pooling their info I don't think an AntagHUD amounts to much more than QOL for ghost mains and especially for those players who die during the round that want to sit out and know if their killer was valid or not.

### Full disclosure: This is bounty code!

Forces DNR for ghosts when they use the verb if they are revivable.
Adds a var to observers for tracking if they have confirmed unlocking.
Checks that var when making the list of ghost candidates for things like mob offers and dynamic mid-round spawns, and also for static spawners to prevent rejoining the round. Including anything that uses `obj/effect/mob_spawn/attack_ghost()` and Swarmer Beacons and Dragon Rifts.
Saves a game log when used.

Before unlock:
hd3vdpRJFo
qfkYyvYhgd

After unlock:
v4HPsQzd8t
Plus all the antag HUD icons of course.

## Why It's Good For The Game

QOL good period, but also..

Adds a highly requested feature to the game that most popular codebases have which properly integrated seems be no issue for their admins. Although I can only judge that by the fact the feature exists and is still enabled for player use.

Better observing for those who only want to spectate. Happier ghosts 👻

## Changelog
:cl:
expansion: Introduced the "Unlock Antag HUD" ghost verb. Once used adds all antag data to your Data HUDs and makes all antags appear as Ghost-Visible Antagonists in the Orbit menu. It also makes you unable to participate in the rest of the current round either from revival (forces DNR), ghost role offers, or ghost spawners.
config: Added flag/disable_ghost_unlock_antag_hud for disabling the new verb if needed without code changes.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
can you make enabling this a config?

**Comment by Wayland-Smithy** [comment]
> can you make enabling this a config?

Sure enabled or disabled by default?

**Comment by Fikou** [comment]
> > can you make enabling this a config?
> 
> Sure enabled or disabled by default? as in when the config option is commented out. `flag/disable_ghost_antag_hud_unlock` vs `flag/allow_ghost_antag_hud_unlock`

enabled by default sounds good, so headmins dont have to turn it on manually and it can be turned off if theres too much abuse

File: code/modules/mob/dead/observer/observer.dm
Comment (Fikou) [inline code/modules/mob/dead/observer/observer.dm:785]: make spawners like golems check for this as well

**Comment by TemporalOroboros** [comment]
Should this also cut them off from ghost chat or will we have ghosts drawing straws for who gets to be the lookout?

**Comment by Wayland-Smithy** [comment]
I don't think that will be nearly as much of an issue as it might seem when currently you can orbit admins as they click through the secrets, listen for traitor/cult/holo para chat, and even ask already dead team antag valids to sus out the rest of the antags while not giving up the ability to rejoin the round. At some point it comes down to taking personal responsibility when it comes to not acting on meta knowledge your character should not have. If anything this might make it easier to spot the troublemakers if they are obviously seeking info from the "lookouts" and then beelining to the antags or blathering on the radio once defibbed or whatever.

**Comment by MrStonedOne** [comment]
Agreed, i'd even be willing to support a timer once you disable it before you could use ghost spawns again, bay has this, 30 minutes after you disable it you can ghost spawn again, This can be mostly helpful for really long rounds that you died earlier in the round for.

**Comment by Wayland-Smithy** [comment]
How about making it use a cooldown, could also be configable I suppose, that's 30 minutes once you use the Unlock verb? 

Then say you have `antag_sight_unlocked` you would also need to be cooling for spawns to be blocked.
There's potential to use a timer also so we can `to_chat` them when their cooldown has ended. Either way I think we'd need 1 more var for tracking if they are cooling and unable to spawn separate.

**Comment by Kylerace** [comment]
im not convinced this is good, this just sounds like a potential meta info leak for little gain. i dont accept happy ghosts as a valid reason, there are ways of giving ghosts something to do other than watch the antag. can you explain why this would improve the game?

**Comment by Iamgoofball** [comment]
:-1: you guys thought metagaming was bad before lmao, holy shit this is gonna be nuts

**Comment by Arkatos1** [comment]
Antag HUD is not exactly a brand new invention and servers that have it do not exactly have more problems with metagaming than we already do. So I really doubt the opinion of some here who are hearing about this the very first time and think this will lead to some meta armageddon and servers on fire.

**Comment by Wayland-Smithy** [comment]
> this lets out information that should not be readily available, it simply shouldnt.

Already you are on shaky ground when the amount of info being "let out" is not even that huge once, and you seem to agree.

> ghosts are certainly able to currently figure out who is antag and who isnt better than a live player can, that is indisputable. but making it easier is not the right direction.

Why is it not the right direction? How are the ghosts that can't join the round going to cause *more* problems than they already can? Like I said before if antags who have not done a single thing to make them valid are hunted and killed from a pair of shitters in their VC it should be super obvious. If not then the info used must have been the kind which is already available to ghosts as is and thus it makes no difference.

I hate to even bring this up but it sounds like you are making a "slippery slope" argument as if AntagHUD will open the flood gates to some barley held back meta ghosting horde and that would eventually lead to observers being deleted altogether.

> it gives the game little except more work for admins

It's not as if dchat does not ask the cool admins for info already and are obliged more times than not. That's just like your opinion and I don't appreciate the seeming kneejerk DNM. Can a rep from the admin team please weigh in?

**Comment by Iamgoofball** [comment]
@Arkatos1 
>Servers that already have it do not exactly have more problems with metagaming than we already do

can you prove this bold claim

**Comment by Iamgoofball** [comment]
@Wayland-Smithy 
The admin team has no control here. This is the codebase's domain. Separation of codebase and server is absolute.

**Comment by Wayland-Smithy** [comment]
@Iamgoofball then why are you guys making admin work arguments?

**Comment by Iamgoofball** [comment]
github posters have a mind of their own

**Comment by Wayland-Smithy** [comment]
Also only a sith deals in absolutes. You don't think even a headmin **opinion** could have value here?

**Comment by Iamgoofball** [comment]
oranges is the only absolute in this codebase, the headmins have no power here

**Comment by Wayland-Smithy** [comment]
It's not about power goof, it's about putting misconceptions to bed.

**Comment by Iamgoofball** [comment]
what "misconception" are you claiming here bud, spell it out

**Comment by Wayland-Smithy** [comment]
> what "misconception" are you claiming here bud, spell it out

Okay bud. Since your only reason for asking for such could possibly be from a lack of reading comprehension I will try my best to use simple language.

> 👎 you guys thought metagaming was bad before lmao, holy shit this is gonna be nuts

Sounds like you are claiming metagaming is already a major issue. Do you have any evidence of this? Like from the public bans DB even?

**Comment by Arkatos1** [comment]
> 
> 
> @Arkatos1
> 
> > Servers that already have it do not exactly have more problems with metagaming than we already do
> 
> can you prove this bold claim

Nothing more than being part of the community for years that had AntagHUD and had very little issues with it. Someone metagaming will be metagaming even without it existing, the difference is that AntagHUD uses are well logged.

**Comment by Wayland-Smithy** [comment]
If removing the config changes temporarily will make this PR better suited for a Test Merge let me know, and we can let it run for a bit to gain some empirical data which *might* help to shed some light on the actual effects of such a change instead of arguing what ifs and anecdotes from a bygone era.

**Comment by Iamgoofball** [comment]
people are metagaming to this very day, there's a ban appeal up for someone metagaming right now 

metagaming isn't a bygone era, it's a cheating issue that's happening right now, and has been happening a lot recently, look at the perma log, look at ban appeals, come on

**Comment by Wayland-Smithy** [comment]
> metagaming isn't a bygone era

Your arguments that an AntagHUD has contributed in the past are.

**Comment by Iamgoofball** [comment]
??????????????????? where the fuck did i argue that we haven't had antaghud available to the players in the past

**Comment by Wayland-Smithy** [comment]
> ??????????????????? where the fuck did i argue that we haven't had antaghud available to the players in the past

Precisely. I was pointing out the multiple times it's been said how "we had antagHUD in the past and it did not go well" not anything you may have posted specifically. I didn't quote or mention you did I?

Also that's a pretty bad faith interpretation of my comment imo. Please get some rest goof and come back refreshed cause I'd really be disappointed if we became rivals over something so petty ❤️

**Comment by Iamgoofball** [comment]
chrome_nWyaypexQp
chrome_fkfltBQiMx
chrome_wvIYwfKyRM

these are just a few examples from the last few months

here's some additional examples of active metacommers this year!
https://tgstation13.org/phpBB/viewtopic.php?f=34&t=29202&p=601856&hilit=metacomms#p601856
https://tgstation13.org/phpBB/viewtopic.php?f=34&t=28915&p=597686&hilit=metacomms#p597686

metacomming and cheating in ss13 is unfortunately still a very real problem and it's absolutely the worst thing to deal with, i'd really rather we not risk making the problem worse by lowering the bar for entry when it comes to cheating

it doesn't matter if there's a log and we catch them afterwards because of it, the antag was still victimized by a cheater and it's simply not cool

**Comment by Iamgoofball** [comment]
it's the main issue with VAC in Valve games

sure, VAC will catch a cheater, but not until they've ruined potentially hundreds of games because it can only catch a cheater in post

imagine if you could live spectate your buddy's CS:GO competitive matches and tell them where all the enemies were

**Comment by Wayland-Smithy** [comment]
Compared to the rest of the data what proportion would you say is metacomms related? Also could the underlying **raw** data not be watched by any concerned admins during a test merge to see if there is a significant change before a full merge if that is a concern? Data is king.

> the antag was still victimized by a cheater and it's simply not cool

They can however be given an antag token in their notes after the offending party has been dealt with, but seeing as we won't even give Wizards a second chance for all but the most egregious admin mistakes I doubt that'll happen.

But now we are getting into policy again which goes against the "absolute" separation 🙊

**Comment by Wayland-Smithy** [comment]
> imagine if you could live spectate your buddy's CS:GO competitive matches and tell them where all the enemies were

**Comment by Wayland-Smithy** [comment]
oops wrong button,

**Comment by Wayland-Smithy** [comment]
> imagine if you could live spectate your buddy's CS:GO competitive matches and tell them where all the enemies were

The CS:GO example is a bad once since when you die you are only allowed to spectate your own team while you are still obviously talking to them via "meta comms" which there is no rules or culture against.

In order to supply that info you literally need to be cheating and yes the matches you are in while doing so are ruined for everyone else but the competitive mode survives despite this with even less tools to make it better than we have.

**Comment by LemonInTheDark** [comment]
I agree with kyler. Additionally, this will lead to lower midround spawns as people will get bored, hit the button, and then be unable to get back into the game

**Comment by LemonInTheDark** [comment]
> That's just like your opinion and I don't appreciate the seeming kneejerk DNM. Can a rep from the admin team please weigh in?

Don't care about the admin meme (something something seperation), but we're maintainers, it's our job to have opinions. Expecting otherwise is silly.

On that note, kyler asked why you thought this was good for the game. I'd also like to hear more on this, since the current reasoning doesn't really go much past other servers have it, and it'll give ghosts easier access to info they already have. 
Frankly, even outside of metagaming and murderboning memes, I don't think giving ghosts instant info about that stuff is helpful for the fun of the ghosts, kinda spoils the story a bit at least in theory.

**Comment by Wayland-Smithy** [comment]
> Additionally, this will lead to lower midround spawns as people will get bored, hit the button, and then be unable to get back into the game

I doubt it since the logic that checks pop for procing the dynamic injections will not be affected at all by this, and not all ghosts are always candidates anyway based on prefs plus if we do see it become a problem to the determent of mid-round spawns it can be disabled via config or the spawn blocking aspect reexamined at that time.

> Don't care about the admin meme (something something seperation), but we're maintainers, it's our job to have opinions. Expecting otherwise is silly.

I'm all for maintainer opinions, but when they are based on the **perceived** impacts this PR **might** have on the admin team which afaik the poster of the opinion in question is not I think it's fair to cast my doubts and propose seeking confirmation. From the standpoint of "power user" **_players_** only of course with no undue sway whatsoever over the maintainers; Just in-case I see this screenshotted on coding gen for the absolute meme.

> On that note, kyler asked why you thought this was good for the game.

QOL is good. Need I explain why?

> kinda spoils the story a bit at least in theory.

Like [Alfred Hitchcock talked about in this famous passage](https://www.goodreads.com/quotes/728496-there-is-a-distinct-difference-between-suspense-and-surprise-and), suspense with the audience knowing the secrets, is often if not more enjoyable than just the surprise of learning the truth. Think Knives Out vs Murder on the Orient Express different yet similar in this regard. At the end of the day it's only an option and options give the players more agency to get the most enjoyment possible out of the game.

My biggest take away from all these arguments is this: The QOL for Observers is not worth making what now might take a minute or so for a dedicated meta ghoster to do with auto observe which works for all but the most stealthy traitors and revs with no contraband anyway take relatively no time so screw 'em all.

**Comment by Kylerace** [comment]
sorry for taking a while i got busy today.

i dont accept qol as enough reason, i dont value it compared to the downsides. qol implies an easier way to do things, but i disagree with the idea that this should be easier in the first place. you have not given any reason as to why this PR has any benefit besides making something easier which i dont believe should be easier. as for asking admins, the only admin opinions i have seen so far on this have been completely against it. 

>That's just like your opinion and I don't appreciate the seeming kneejerk DNM

i applied the DNM tag because i don't think its right to close something like this without the authors defense of it at the very least. youve given your response and i still see no merit in this so i will close this PR.

**Comment by MrStonedOne** [comment]
> The admin team has no control here. This is the codebase's domain. Separation of codebase and server is absolute.

The moment maintainers have invoked the admin work argument, it became the admin's domain.

**Comment by Iamgoofball** [comment]
hm, fair

**Comment by Wayland-Smithy** [comment]
> i dont accept qol as enough reason

That's your prerogative then, as it seems like you are not interested in any debate on the matter despite your claims of the DNM label supposedly setting the stage for it.

> the only admin opinions i have seen so far on this have been completely against it.

Could you kindly point them out to me? I must have missed those comments perhaps they are hidden.

> i dont accept qol as enough reason, i dont value it compared to the downsides. 

Fair enough, "screw all ghosts since 0.1% of them might be bad faith actors" it is then.

> qol implies an easier way to do things, but i disagree with the idea that this should be easier

It sounds like we simply disagree on the definition of Quality of Life which is understandable since for videogames at least there is some debate of whether or not not such features should effect gameplay. Even though this PR is exactly that "an easier way to do things" even if you think a thing should not be easier does not make my proposed changes here any less QOL imo. It's not like I am trying to masquerade a tator tot buff as QOL for example.

**Comment by Wayland-Smithy** [comment]
> you have not given any reason as to why this PR has any benefit besides making something easier

As I'm sure you know there are no other reasons besides QOL for Observers, and highly requested QOL by the players at that. Which the benefits of vastly dwarf the perceived potential detriments I've outlined in the body and other comments. For me to try and claim any other reason would be disingenuous and I'm sure you'd have been very happy to call me out for something like that and use it to close the PR also.

> At the end of the day it's only an option and options give the players more agency to get the most enjoyment possible out of the game.

**Comment by MrStonedOne** [comment]
> even if the people who abuse it get banned it gives the game little except **more work for admins**

image

>Additionally, this will lead to lower midround spawns as people will get bored, hit the button, and then be unable to get back into the game

If the presence of midround spawns is gonna get used to gimp other ghost experiences, I will remove and disable all mid round spawns server side. This is not a game about a singular experience, **stop trying to tell people how to enjoy the game**. 

That argument is not going to fly with me.

As long as we ban twitch streaming, we have to provide people who enjoy that kind of observer experience something to compensate. There is zero reason for people to have this disdain for observers. As far as im concerned, they are valid players of the game, even if they never play, and **absolutely** a playerbase cohort to target with better experiences.

And As long as we expect dead people who come back to not use meta information they gained from ghost chat, i don't really think we can argue for the removal of meta information at the cost of preventing better experiences to players.

This is a **role playing** game. Not a video game. Stop trying to treat it as a video game. SS13 has and will always have more in common with table top rpgs then video games. *This is by design*.

Better ideas for a compromise:

Allow tators to opt-out of ghost antag hud on a round by round bases so somebody who is going stealth and doesn't want to risk it can choose to do so.

Add a seperate dchat for antag hud users. (removing dchat entirely is off the table)

**Comment by Wayland-Smithy** [comment]
> Add a seperate dchat for antag hud users.

I will look into creating a "elevated dchat++" channel which will allow the hud users to only send dchat messages to each other on top of normal dchat, with +ADMIN moderation too, and perhaps incorporate another warning into the alert text.

> (removing dchat entirely is off the table)

For sure, that is not something I was ever going to do in this PR anyway.

**Comment by MrStonedOne** [comment]
I should point out, that I don't think any of those things are needed, they are just tokens thrown out to the maintainers.

There are a lot of unlogged ways to see that somebody is an antag as a ghost, some of them a good chunk of admins don't even know about. (like using observe to see somebody's hud, giving you info about any uplink action buttons they have, any tator items they have in their backpack/pockets/etc)

Stopping what I think is a worthwhile feature in the face of that under fears of metagaming that can already happen doesn't track with me.

And this entire issue boils down to people who aren't chronic observers, and/or have the very button they want to deny to players, not understanding this pr or why chronic observers would want it, and using "rationalization hamster" arguments to justify their kneejerk reaction against it.

So here is my argument on why this is beneficial:

Ghosts can already see this info, chronically observing ghosts like to know this info, but ghosts currently have to go thru a whole bunch of work hitting observe on every human and checking their hud for action buttons (like uplink implant) and contents for tator items as well as checking around the map for people doing antag things and keeping an eye on chat for details about antags like cultist invocations in say.

Given that ghosts can already see the info, want the info, but has to go thru a complex method to have it, we should give them a better way that leads to the same end point but without making the observer jump through hoops.

Also its logged and this will make admin's word easier.

**Comment by Kylerace** [comment]
remove the ability for antag hud observers to speak in normal dchat and i will lift dnm, having a separate dchat is acceptable. i think its fair to say i was hasty to close due to administration concerns, but i will not allow them to interact with non antag hud players in any way except via attempting to cheat (which the logging can counteract).

>I should also point out, that ghosts can see somebodies inventory using the observe verb.

>They also see all action verbs the person has with the same verb.

i consider that stupid as well but removing something from the game is different than blocking a new feature

**Comment by Iamgoofball** [comment]
this is both a roleplaying game and a video game, seeing as you are not a maintainer it's apparent you accidentally reopened this PR, let me contact a maintainer to fix this issue

**Comment by Wayland-Smithy** [comment]
> i think its fair to say i was hasty to close due to administration concerns

@Kylerace I appreciate you admitting wrong, even with qualifiers, on the internet +respect. Sorry the rest of the quotes are not in chronological order.

>> I should also point out, that ghosts can see somebodies inventory using the observe verb.
>> They also see all action verbs the person has with the same verb.

> i consider that stupid as well but removing something from the game is different than blocking a new feature.

Meaning if you could easily force that change into the game you would have already? Like closing this PR only retroactive. _nice_

> remove the ability for antag hud observers to speak in normal dchat
> i will not allow them to interact with non antag hud players / the round at large in any way

Even if I made that change when would the goalpost stop moving? Most of your last comment leads me to believe never. However you are free to code any of the out of scope ideas after merge, and I vow not to interact with those PRs regardless of my opinion.

After sleeping on it I realize just limiting hud users dchat is barely a stop gap when the rest of the ghosts can still see who they are orbiting. Plus the silent active ghosts will be assumed to be hud users and just asked to orbit POIs or use point by others anyway. Especially in cases like 1 remaining rev head. So now either the total orbiter count needs to be over engineered, or point also disabled, or hud user ghosts made invisible like admins etc.. All things which will add even more maintainability cost.

> i will lift dnm

If not you then could somebody else please and thanks?

File: code/modules/mob/dead/observer/observer.dm
Comment (coiax) [inline code/modules/mob/dead/observer/observer.dm:776]: Can we make the verb itself conditional on the config being enabled? Is that a power we have with Byond?

File: code/modules/mob/dead/observer/observer.dm
Comment (coiax) [inline code/modules/mob/dead/observer/observer.dm:787]: Maybe you shouldn't copy and paste the code.

**Review by coiax — CHANGES_REQUESTED** [review]
It's possible to bypass this flag by playing CTF or some other embodying minigame, since observers are different objects once you've inhabited a body.

(Code review only, I am undecided on whether to support this.)

File: code/modules/mob/dead/observer/observer.dm
Comment (MrStonedOne) [reply code/modules/mob/dead/observer/observer.dm:776]: > Is that a power we have with Byond?

yes, you make it a proc instead of a verb and then add the proc to the mob's verb list during initialize.

to complete it, you also have to add code to the config item so if admins reload config mid round or vv the config, the verb is added to all existing ghosts.

Comment (Wayland-Smithy) [reply code/modules/mob/dead/observer/observer.dm:776]: I could if it was a proc with `verbs +=` on init I suppose.

File: code/modules/mob/dead/observer/observer.dm
Comment (Wayland-Smithy) [reply code/modules/mob/dead/observer/observer.dm:787]: Again just requires another proc to abstract out that logic from inside of the DNR verb.

**Comment by Wayland-Smithy** [comment]
AntagHUD unlock is now tracked via `player_details` and re-applied to ghosts during `Login()` so CTF or any other (un)sanctioned method of spawning cannot revert the unlock stat. Additionally I added a readout for it on the Player Panel so admins can quickly see if some spawn shenanigans took place.

**Comment by Mothblocks** [comment]
> remove the ability for antag hud observers to speak in normal dchat and i will lift dnm, having a separate dchat is acceptable

thats a pretty strange requirement imho, dead chat is the communal salt chamber

**Comment by Cyprex** [comment]
> Ghosts can already see this info, chronically observing ghosts like to know this info, but ghosts currently have to go thru a whole bunch of work hitting observe on every human and checking their hud for action buttons (like uplink implant) and contents for tator items as well as checking around the map for people doing antag things and keeping an eye on chat for details about antags like cultist invocations in say.

> Given that ghosts can already see the info, want the info, but has to go thru a complex method to have it, we should give them a better way that leads to the same end point but without making the observer jump through hoops.

Yes, and making the information reliable(traitor items in backpack can be obtained from a traitor) and easily accessable makes this much, much easier to abuse. You pretty much said this yourself here.

**Comment by Watermelon914** [comment]
> > Ghosts can already see this info, chronically observing ghosts like to know this info, but ghosts currently have to go thru a whole bunch of work hitting observe on every human and checking their hud for action buttons (like uplink implant) and contents for tator items as well as checking around the map for people doing antag things and keeping an eye on chat for details about antags like cultist invocations in say.
> 
> > Given that ghosts can already see the info, want the info, but has to go thru a complex method to have it, we should give them a better way that leads to the same end point but without making the observer jump through hoops.
> 
> Yes, and making the information reliable(traitor items in backpack can be obtained from a traitor) and easily accessable makes this much, much easier to abuse. You pretty much said this yourself here.

It also makes it much easier to catch because it's actually logged.

**Comment by Cyprex** [comment]
That helps but the far bigger issue is still identifying that two players are working together, which will often go unnoticed

**Comment by Watermelon914** [comment]
> That helps but the far bigger issue is still identifying that two players are working together, which will often go unnoticed

This narrows the possible culprits. If someone is determined to metagame, it won't change the fact that they'll do so, you'll just be impacting user experience of ghosts.

**Comment by Kylerace** [comment]
im going to abstain from this pr now. convince the other maintainers

**Comment by Timberpoes** [comment]
With Kyler stepping out, I am going to be closing the PR. I basically shared his concerns and I will outline my reasons for closure below:

I don't want observers to have perfect knowledge. I don't believe it contributes anything to the narrative or story of the shift for observers to know everything with no room for error in knowledge.

Regarding assertions that some whodunit mystery novels are inherently more fun to read because the audience know exactly who the bad guys are from the outset, I do not believe this directly translates to SS13 observers and ultimately even if it did, it is not the kind of storytelling device I wish to insert into the game.

Sometimes observers will discover who the baddies are through watching the actions of other players. This is the kind of thing I believe makes for better stories than having the mystery spoiled from the outset.

I prefer for the shift to tell its own story as opposed to the game having spoiler buttons from the outset. Anything an observer has access to you should expect a temporarily dead player to have access to as well. I'd want to limit the amount of spoilers the temporarily dead are exposed to while balancing the overall experience of observers. I believe this PR tips the scale too much in favour of observer spoilers to the detriment of the other side.

To me it's about balancing the ability for observers to witness a story versus the ability for players to tell it, while carefully managing the risk elements of exposing to-be-revived players to the information that observers get. Being spoiled while actively participating ruins the entire point of the game vis-a-vis chaotic paranoia-laden murder mystery with horror undertones. Resolving this impasse is unreasonable without ruining other elements of the game.

There is no oven-ready solution to my concerns. It comes down to the fact that this PR seeks to introduce something I believe detrimental to the method of storytelling that tg's brand of SS13 thrives on. Any attempts to address my concerns are likely to render moot the underlying point/objectives of the PR or would subtract from other areas of the game more than this would add.

tl;dr - I am opposing steps taken to make this information easier to access or more mainstream for observers. I believe the detriments to making this information more accessible outweigh any benefits to observers having perfect information.

As a result, I am closing this PR as it's not something I wish to see merged. If another maintainer desperately wants to see this feature added to the game, ping me on Discord and we'll see if I'm willing to nudge my position a little bit to let this through.

**Comment by Wayland-Smithy** [comment]
> Regarding assertions that some whodunit mystery novels...

No need to argue about literature and how it is clearly different from a SS13 story generator. I just wanted to give LemonTato a concrete example to the contrary of their assertion which I suspect they only made in passing because someone else volunteered to slap on the DNM.

This part of your reasons too right?

> In my eyes if you're exposing this level of information to observers, you might as well just allow players to bring up the end of round summary at any moment and see all the antags, all their objectives and whether they've greentexted any of them.

I like how the fact they are blocked from coming back into the round means nothing in your eyes too because they can still chat, but that simply is not true. How many "kind" ghosts with it unlocked are going to *really* volunteer to play antag announcer and be taken advantage of by the ghosts who don't want to make the same sacrifice? Also claiming they might as-well see "the end of round summary at any moment" is hyperbole and you know it. This PR is about AntagHUD not a crystal ball that can see into the future.

> It basically ruins the ability for the shift to tell a story by spoilering the conclusion upfront - And if you don't care about the conclusion, tough shit because dead chat is already shift spoiler central.

Can't DChat just be muted by those who don't want to be spoiled too? Not sure how that isn't already a problem for you that needs fixing, but okay the saga continues.

**Pull request closed without merging**
